src/makeCalculator.js

```diff
@@ -4,7 +4,46 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        console.log('Cannot divide by zero');
+        return;
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
+      }
+
+      return this;
+    },
+  };
+
+  return calculate;
```

Comment:
The variable name `calculate` does not follow the naming convention described in the checklist. Since the function is named `makeCalculator`, the returned object should be named `calculator` to better reflect its purpose.